core/trino-spi/src/main/java/io/trino/spi/predicate/SortedRangeSet.java

```diff
@@ -842,11 +842,26 @@ public String toString(ConnectorSession session)
 
     private String formatRanges(ConnectorSession session, int limit)
     {
-        return Stream.concat(
-                IntStream.range(0, min(getRangeCount(), limit))
+        if (isNone()) {
+            return "{}";
+        }
+        if (getRangeCount() == 1) {
+            return getRangeView(0).formatRange(session);
+        }
+        if (limit < 2) {
+            return "span=" + getSpan().toString(session);
```

Comment:
is this reachable?

Reply:
it's used in subsequent commit `Use Domain#toString with limit for DynamicFilterDomainStats`

Reply:
so it should be added there. however, i would prefer NOT to have this at all. As commented elsewhere, `limit=1`  servers as a magic constant to invoke "DF friendly toString representation".
I'd prefer to keep the toString here as simple as possible.
If DF printing requires some specialized formatting, let it be handled in the DF-related code.

Reply:
I already got rid of the `limit=1` case, I'm using `limit=2` now. We have consistent behaviour now for all values of limit.
I kept the addition of `limit` parameter and its usage in separate commits for cleanliness, otherwise we would be mixing SPI code and DF specific code changes in the same commit.
Earlier DF code was performing simplification and relying on toString to be implemented a certain way. Since there was no contract about that behaviour, we went from
```
"dynamicFilterDomainStats" : [ {
        "dynamicFilterId" : "df_370",
        "simplifiedDomain" : "[ [[2451546, 2451905]] ]",
        "rangeCount" : 3,
        "discreteValuesCount" : 0,
        "collectionDuration" : "2.34s"
      } 
```
to
```
"dynamicFilterDomainStats" : [ {
        "dynamicFilterId" : "df_332",
        "simplifiedDomain" : "[ SortedRangeSet[type=bigint, ranges=1, {(2451546,2451905)}] ]",
        "rangeCount" : 3,
        "discreteValuesCount" : 0,
        "collectionDuration" : "75.04ms"
      } ]
```
Now it looks confusing because range count is repeated. It's not a big deal for the queryinfo JSON but I think we wouldn't want an output like that in EXPLAIN ANALYZE or the web UI.
So we need a way to print Domain in a compact and clean way. Tweaking the toString and exposing limit parameter seemed reasonable as it's close to meeting the requirement and we were relying on unspecified behaviour from it anyway.
Not agreeing to a contract here would just mean a repetition of similar code elsewhere.